core/src/com/unciv/logic/city/CityConstructions.kt

```diff
@@ -53,6 +54,9 @@ class CityConstructions {
     var productionOverflow = 0
     val queueMaxSize = 10
 
+    // Maps cities to the buildings they received
+    val freeBuildingsProvidedFromThisCity: HashMap<String, HashSet<String>> = hashMapOf()
```

Comment:
Serialized. You sure it isn't preferable to re-calculate on game load?

Reply:
That wouldn't work, because we have to remember somewhere which buildings we added to which cities. If there are two cities, one of which already contained a broadcast tower and one which was added through the CN tower, we have to differentiate these somehow

Reply:
Yeah, I was wondering. So I build a BT in 2 of 5 cities, then the CN. All 5 cities would show a BT in the city screen as built (right?). Can it show which are the free ones?

I now lose CN -> I'll still have the two BT I built first, only the rest is gone? Is it supposed to work like that? I guess you would know. So - OK something needs persistence, understood.